### PR TITLE
Handle situations where IPv4 multicast is not available

### PR DESCRIPTION
**Pull request by jeremyevans**
Also, switch `skip` to `pend` so tests pass outside of Ruby repo.

Fixes [Bug #13864]

### CODE REVIEW

File: test/rinda/test_rinda.rb [hunk 1]
```diff
@@ -402,7 +402,7 @@ def test_cancel_01
   end
 
   def test_cancel_02
-    skip 'this test is unstable with --jit-wait' if defined?(RubyVM::JIT) && RubyVM::JIT.enabled?
+    pend 'this test is unstable with --jit-wait' if defined?(RubyVM::JIT) && RubyVM::JIT.enabled?
```
Comment (olleolleolle) [inline test/rinda/test_rinda.rb]: Can this be "omit" and still work? Are we expecting the test ever to become stable with jit-wait? 

If not, perhaps skip->omit is another way of saying skip?

(This is just me pattern-matching on this change, which was done in another standard library gem a while ago.)

Super-nice fix, with the mixin module taking care of the testing for support.

Comment (kou) [reply test/rinda/test_rinda.rb]: In test-unit, `omit` is the right alternative of `skip` of minitest.
`pend` is for a case that will be fixed later.
`omit` is for a case that is never ran.

Comment (jeremyevans) [reply test/rinda/test_rinda.rb]: OK, I switched `pend` to `omit`.

**Comment by jeremyevans** [comment]
@seki I rebased this against the master branch.  Could you please review when you have time? Alternatively, @mame or @hsbt, could you review this?

**Comment by kou** [comment]
Thanks!

**Pull request merged**